js/list.js

```diff
@@ -0,0 +1,204 @@
+import { initializeApp } from "https://www.gstatic.com/firebasejs/10.1.0/firebase-app.js";
+import { getDatabase, onValue, ref, set, get, child } from "https://www.gstatic.com/firebasejs/10.1.0/firebase-database.js";
+
+var firebaseConfig = {
+    apiKey: "AIzaSyBZ33dDbUj7p5n8k5aQivuDyPqVx1OU1vQ",
+    authDomain: "task1---events-webpage.firebaseapp.com",
+    databaseURL: "https://task1---events-webpage-default-rtdb.firebaseio.com",
+    projectId: "task1---events-webpage",
+    storageBucket: "task1---events-webpage.appspot.com",
+    messagingSenderId: "642153429925",
+    appId: "1:642153429925:web:24b7fc5f171016582cc64c",
+    measurementId: "G-9N01SC4T9N"
+};
+
+const app = initializeApp(firebaseConfig); //To initialize Firebase
+const db = getDatabase(app);
+const newBtn = document.getElementById("new-event-btn");
+const searchBar = document.getElementById("searchBar");
+const list = document.getElementById("sort-options");
+
+var totalRecord = 0;
+loadEvents(); //load the events from database to the webpage
+
+
+//*****************EVENT LISTENERS*****************/
+//NEW+ button functionality
+newBtn.addEventListener("click", newEvent);
+function newEvent() {
+    set(ref(db, 'eventToEdit/'), { //when user clicks new - no events to edit so make sure eventToEdit is set to none 
+        id: "none",
+        name: "none",
+        description: "none",
+        date: "none"
+    });
+    window.location.href = "form.html";
+}
+
+//Search bar functionality 
+searchBar.onkeyup = () => {
+    let searchName = document.getElementById("searchBar").value.toUpperCase();
+    let cards = document.getElementsByClassName("card");
+    //iterate through all cards and hide cards that don't match the search query
+    var i;
+    for (i = 0; i < cards.length; i++) {
+        let name = cards[i].textContent.toUpperCase(); //or name = cards[i].innerText;
+        if (name.indexOf(searchName) > -1) { //check if searchName is substring of card name
+            cards[i].style.display = "";
+        } else {
+            cards[i].style.display = "none";
+        }
+    }
+}
+
+//Sorting drop-down list functionality
+list.onchange = () => {
+    clearEvents();
+    loadEvents();
+}
+
```

Comment:
lets update this logic by:
- create 2 arrays
   - one for all data
   - one for filtered data
- by default (the 2 arrays are equal)
- on search or sort, update the filteredArray and re-build the cards

Reply:
done